usr/src/cmd/zfs/zfs_iter.c

```diff
@@ -391,7 +392,7 @@ zfs_for_each(int argc, char **argv, int flags, zfs_type_t types,
 		}
 
 		while (sortcol) {
-			if (sortcol->sc_prop >= ZFS_PROP_TYPE &&
+			if ((int)sortcol->sc_prop >= ZFS_PROP_TYPE &&
```

Comment:
this change seems wrong, because both of these are zfs_prop_t's.  If it's necessary, why don't we need to cast it on the next line?